contracts/hydro/src/contract.rs

```diff
@@ -715,71 +736,89 @@ fn vote(
         // Get any existing vote for this sender and reverse it- this may be a vote for a different proposal (if they are switching their vote),
         // or it may be a vote for the same proposal (if they have increased their power by locking more and want to update their vote).
         // TODO: this could be made more gas-efficient by using a separate path with fewer writes if the vote is for the same proposal
-        let vote = VOTE_MAP.load(
+        let vote = VOTE_MAP.may_load(
```

Comment:
Apart from this load()->may-load() change, it's just an indentation change.
The None "arm" is newly added code.